sale_delivery_date/models/sale_order_line.py

```diff
@@ -41,13 +41,21 @@ def _prepare_procurement_values_commitment_date(self, res):
         so we will be able to ship the goods to the customer in respect to the
         `commitment_date`.
         The 'date_deadline' is already set to the 'commitment_date' (that should
-        already fit the partner's time window if any).
+        preferably fit the partner's time window if any).
         """
-        # TODO
-        # 1) commitment_date - security_lead = order ready to be shipped {1}
-        # 2) {1} - workload
-        # 3) while {2} isn't a working day, remove 1 day
-        # 4) apply cutoff (with `keep_same_day` param)
+        # 1) compute the date when the transfer should be ready to be shipped
+        date_transfer_done = res["date_deadline"] - timedelta(
+            days=self.order_id.company_id.security_lead
+        )
+        res["date_planned"] = date_transfer_done
+        # 2) Find the first date in the WH calendar (by going back in the past)
+        calendar = self.order_id.warehouse_id.calendar_id
+        if calendar:
+            res["date_planned"] = calendar.plan_days(
+                -1, res["date_planned"], compute_leaves=True
+            )
```

Comment:
is this really working? :open_mouth:

Reply:
I didn't take time to test it in a real use case or TU, but after checking the implementation of this method and a quick test in a shell, seems to work as expected yes:
```python
>>> from odoo import fields
>>> calendar = env["stock.warehouse"].browse(1).calendar_id  # Weekend are days off in this calendar
>>> monday = fields.Datetime.from_string("2022-04-04")
>>> calendar.plan_days(-1, monday, compute_leaves=True)
datetime.datetime(2022, 4, 1, 11, 0)  # Friday 1st April
```